opennms-doc/guide-install/src/asciidoc/text/time-series-storage/newts.adoc

```diff
@@ -0,0 +1,148 @@
+
+TIP: To avoid unwanted updates disable the _Cassandra_ repository on _DNF/YUM_ based distributions or use `apt-mark hold cassandra` on _APT_ based distributions.
+
+By default using the `${OPENNMS_HOME}/bin/newts init` command the samples keyspace is created with the link:https://docs.datastax.com/en/dse/5.1/dse-arch/datastax_enterprise/dbInternals/dbIntHowDataMaintain.html#dbIntHowDataMaintain__stcs-compaction[Size Tiered Compaction Strategy (STSC)].
+_STSC_ is not optimal for time series data and requires additionally 50% guaranteed free disk space on the partition with SSTables to complete compactions.
+For this reason we will describe how to set up samples table with the link:https://docs.datastax.com/en/dse/5.1/dse-arch/datastax_enterprise/dbInternals/dbIntHowDataMaintain.html#dbIntHowDataMaintain__twcs[Time Window Compaction Strategy (TWCS)] which has an overhead approximately around 10-20%.
+
+
+==== Objectives
+
+* Configure _{opennms-product-name}_ to use an existing _Cassandra_ cluster
+* Initializing the _Newts_ keyspace using link:https://docs.datastax.com/en/dse/5.1/dse-arch/datastax_enterprise/dbInternals/dbIntHowDataMaintain.html#dbIntHowDataMaintain__twcs[TWCS] for the samples table
+* Verify time series data is stored and can be accessed
+
+==== Before you begin
+
+* A running instance of _{opennms-product-name}_ running on Linux
+* Working data collection and response time metrics from _Collectd_ and _Pollerd_
+* _Cassandra_ cluster with access to the _Cassandra_ client port `TCP/9042`
+
+WARNING: It is currently not supported to initialize the _Newts_ keyspace from _Microsoft Windows Server_ operating system.
+         _Microsoft Windows_ based _Cassandra_ server can be part of the cluster, but keyspace initialization is only possible using a _Linux_ operating system.
+
+
+
+[[gi-configure-opennms]]
+==== Configure {opennms-product-name} to use Newts
+
+
+// No section numbers for step-by-step guide
+:!sectnums:
+
+===== Step 1: Configure Cassandra endpoints, keyspace and time series strategy
+
+[source, shell]
+----
+cat << EOF | sudo tee /opt/opennms/etc/opennms.properties.d/timeseries.properties
+# Configure storage strategy
+org.opennms.rrd.storeByForeignSource=true<1>
+org.opennms.timeseries.strategy=newts<2>
+
+# Configure Newts time series storage connection
+org.opennms.newts.config.hostname={cassandra-ip1,cassandra-ip2}<3>
+org.opennms.newts.config.keyspace=newts<4>
+org.opennms.newts.config.port=9042<5>
+EOF
+----
+
+<1> Associate time series data by the foreign ID instead the database generated node ID
+<2> Set time series strategy to use `newts`
+<3> Host or IP addresses of the _Cassandra_ cluster nodes, can be a comma separated list
+<4> Name of the keyspace which is initialized and used
+<5> Port to connect to _Cassandra_
+
+===== Step 2: Initialize the _Newts_ schema in _Cassandra_
```

Comment:
If this is what we're recommending to use for the default schema, should we not just update `newts init` to use this instead?

Reply:
That's another valid approach. I envision that particular enhancement as follow:

Request not only the replication factor but also the desired retention (TTL), from there, calculate the window settings accordingly and generate the schema. Now, I think that would be in the Newts code base, right? But, without having those changes in place, the documentation should be something useful to have.

Reply:
> If this is what we're recommending to use for the default schema, should we not just update `newts init` to use this instead?

that would be my next issue and investigate how to change it there :)